storm-server/src/main/java/org/apache/storm/scheduler/Cluster.java

```diff
@@ -56,6 +56,7 @@
     private static final Logger LOG = LoggerFactory.getLogger(Cluster.class);
     private static final Function<String, Set<WorkerSlot>> MAKE_SET = (x) -> new HashSet<>();
     private static final Function<String, Map<WorkerSlot, NormalizedResourceRequest>> MAKE_MAP = (x) -> new HashMap<>();
+    private static final Function<String, Map<String, Double>> MAKE_MAP_STRING_DOUBLE = (x) -> new HashMap<>();
```

Comment:
I think it would be better to make a "real" function for this, so we can parameterize it with generic params. We can then replace this and MAKE_MAP with the generic version.

```
private static <K, V> Map<K, V> makeMap(String key) {
        return new HashMap<>();
}
```

and at the call site
```
nodeToScheduledResourcesCache.computeIfAbsent(nodeId, Cluster::makeMap)
```